src/include/parser/macro/create_macro.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <unordered_map>
+
+#include "parser/expression/parsed_expression.h"
+#include "parser/statement.h"
+
+namespace kuzu {
+namespace parser {
+
+using default_args = std::vector<std::pair<std::string, std::unique_ptr<ParsedExpression>>>;
+
+class CreateMacro : public Statement {
+public:
+    explicit CreateMacro(std::string macroName, std::unique_ptr<ParsedExpression> macroExpression,
+        std::vector<std::string> positionalArgs, default_args defaultArgs)
+        : Statement{common::StatementType::CREATE_MACRO}, macroName{std::move(macroName)},
+          macroExpression{std::move(macroExpression)}, positionalArgs{std::move(positionalArgs)},
+          defaultArgs{std::move(defaultArgs)} {}
+
+    inline std::string getMacroName() const { return macroName; }
+
+    inline std::unique_ptr<ParsedExpression> getMacroExpression() const {
+        return macroExpression->copy();
+    }
+
+    inline std::vector<std::string> getPositionalArgs() const { return positionalArgs; }
+
+    default_args getDefaultArgs() const;
```

Comment:
Ditto, I wonder if we can just expose raw pointer in getter functions.

Like `vector<string> getDefaultArgNames` and `ParsedExpression* getDefaultArgValue(string argName)`